src/main/webapp/lobby.js

```diff
@@ -24,22 +24,41 @@ channel.bind(PUSHER_CHAT_CHANNEL_NAME, function (data) {
 
 // Fetches list of usernames, appends each username to html list.
 function loadUsernames() {
-    fetch('/user').then(response => response.json()).then((users) => {
-      const userList = document.getElementById('user-list');
-      users.forEach((username) => {
-        userList.appendChild(createUsernameElement(username));
-      })
-    });
-  }
+  // Get current url.
+  let url = window.location.href;
+  let paramString = parseRoomId(url);
+
+  fetch(`/room?roomId=${paramString}`).then(response => response.json()).then((users) => {
+    const userList = document.getElementById('user-list');
+    users.forEach((username) => {
+      userList.appendChild(createUsernameElement(username));
+    })
+  });
+}
 
 // Appends text node to list node, returns list node.
 function createUsernameElement(username) {
-    var node = document.createElement("li");
-    var textnode = document.createTextNode(username);
-    node.appendChild(textnode);
-    return node;
+  var node = document.createElement("li");
+  var textnode = document.createTextNode(username);
+  node.appendChild(textnode);
+  return node;
+}
+
+function parseRoomId(url) {
```

Comment:
It looks like you've copied over this function from join-room.js, which we generally want to avoid doing, since now the same exact code exists in two places. The problem with this is that at a later point, when someone inevitably needs to make changes to this function, they will need to do it in two places. Or worse, they won't realize it's duplicated and make the change in one place but not the other, which can lead to bugs. This is called the "DRY Principle"  (Don't Repeat Yourself):
https://medium.com/@Ialimijoro/the-dry-principle-and-why-you-should-use-it-f02435ae9449

The solution here is to extract parseRoomId() into its own JS file. Something like "common-functions.js", that way we can include other common functions there in the future. You can then include this script in lobby.html and join-room.html, and their respective JS files can both use this common function.

Let me know if you have any questions about this.

Reply:
@pruess-google True, makes sense, fixed!